adbpyg_adapter/adapter.py

```diff
@@ -904,24 +904,28 @@ def __process_adb_edge_df(
             ]
 
             # 5. Map each ArangoDB from/to _key to the corresponding PyG node id
-            from_nodes = et_df["from_key"].map(adb_map[from_col]).tolist()
-            to_nodes = et_df["to_key"].map(adb_map[to_col]).tolist()
+            # NOTE: map() is somehow converting int values to float...
+            # So we rely on astype(int) to convert the float back to int,
+            # but we also fill NaN values with -1 so that we can convert
+            # the entire column to int without any issues. Need to revisit...
+            from_n = et_df["from_key"].map(adb_map[from_col]).fillna(-1).astype(int)
+            to_n = et_df["to_key"].map(adb_map[to_col]).fillna(-1).astype(int)
 
             # 6. Set/Update the PyG Edge Index
-            edge_index = tensor([from_nodes, to_nodes])
-            edge_data.edge_index = torch.cat(
-                (edge_data.get("edge_index", tensor([])), edge_index), dim=1
-            )
+            edge_index = tensor([from_n.tolist(), to_n.tolist()], dtype=torch.int64)
```

Comment:
I'm curious if something like this would be more/less efficient than what's currently proposed:

```py
from_n = torch.tensor(et_df["from_key"].map(adb_map[from_col]).fillna(-1).astype(int).values)
to_n = torch.tensor(et_df["to_key"].map(adb_map[to_col]).fillna(-1).astype(int).values)
edge_index = torch.stack([from_nodes, to_nodes], dim=0)
```

i.e avoiding the use of `tolist()` from pandas

Reply:
addressed in [037fa4d](https://github.com/arangoml/pyg-adapter/pull/31/commits/037fa4d332dca5fc686fcf2f0b691e9fb3aa693a)